src/main/kotlin/com/edibleday/TeaVMTask.kt

```diff
@@ -51,10 +52,11 @@ open class TeaVMTask : DefaultTask() {
         tool.targetDirectory = File(installDirectory)
         tool.targetFileName = targetFileName
 
-        if (project.hasProperty("mainClassName") && project.property("mainClassName") != null) {
-            tool.mainClass = "${project.property("mainClassName")}"
-        } else throw TeaVMException("mainClassName not found!")
-
+        if (mainClass != "") {
```

Comment:
could use `.isNotBlank()`

Reply:
Thanks for noticing! I have never used Kotlin before, I didn't know about this function, I'll use it.